batchflow/models/tf/layers/conv.py

```diff
@@ -1,199 +1,211 @@
+        self.filters, self.kernel_size, self.strides = filters, kernel_size, strides
+        self.padding, self.data_format = padding, data_format
+        self.args, self.kwargs = args, kwargs
 
-    Returns
-    -------
-    tf.Tensor
+    def __call__(self, inputs):
+        axis = 1 if self.data_format == 'channels_last' else 2
+        x = tf.expand_dims(inputs, axis=axis)
+        x = K.Conv2DTranspose(filters=self.filters, kernel_size=(1, self.kernel_size),
+                              strides=(1, self.strides), padding=self.padding, **self.kwargs)(x)
+        x = tf.squeeze(x, [axis])
+        return x
 
-    See also
-    --------
-    `tf.layers.conv2d_transpose <https://www.tensorflow.org/api_docs/python/tf/layers/conv2d_transpose>`_,
-    `tf.layers.conv3d_transpose <https://www.tensorflow.org/api_docs/python/tf/layers/conv3d_transpose>`_
-    """
-    axis = 1 if data_format == 'channels_last' else 2
-    x = tf.expand_dims(inputs, axis=axis)
-    x = tf.layers.conv2d_transpose(x, filters=filters, kernel_size=(1, kernel_size),
-                                   strides=(1, strides), padding=padding, **kwargs)
-    x = tf.squeeze(x, [axis])
-    return x
-
-def conv1d_transpose_nn(value, filters, output_shape, strides,
-                        padding='SAME', data_format='NWC', name=None):
+
+
+@add_as_function
+class Conv1DTransposeNn:
     """ Transposed 1D convolution layer. Analogue of the tf.nn.conv2d_transpose.
 
     Parameters
     ----------
-    value : tf.Tensor
-        input tensor
     filters : tf.Tensor
-        convolutional filter
+        Convolutional filter.
     output_shape : tf.Tensor
-        the output shape of the deconvolution op
+        The output shape of the deconvolution op.
     strides : list
-        the stride of the sliding window for each dimension of the input tensor
+        The stride of the sliding window for each dimension of the input tensor.
     padding : str
         'VALID' or 'SAME'. Default - 'SAME'.
     data_format : str
         'NWC' or 'NCW'. Default - 'NWC'.
     name : str
-        scope name
+        Scope name
+    """
+    def __init__(self, filters, output_shape, strides, padding='SAME', data_format='NWC', *args, **kwargs):
+        self.filters, self.output_shape, self.strides = filters, output_shape, strides
+        self.padding, self.data_format = padding, data_format
+        self.args, self.kwargs = args, kwargs
+
+    def __call__(self, inputs):
+        axis = 1 if self.data_format == 'NWC' else 2
+        inputs = tf.expand_dims(inputs, axis=axis)
+        filters = tf.expand_dims(self.filters, axis=0)
+        output_shape = tf.concat([self.output_shape[:axis], (1, ), self.output_shape[axis:]], axis=-1)
+        strides = self.strides[:axis] + [1] + self.strides[axis:]
+        x = tf.nn.conv2d_transpose(inputs, filters, output_shape, strides,
```

Comment:
Do you need to call `Conv2DTranspose` here?

Reply:
Nope, that is a function that operates directly with `filters`-`Tensor`, not with shapes